Jenkinsfile

```diff
@@ -45,5 +45,19 @@ node('docker && awsaccess') {
         reportName: 'cloudformation-ruby-dsl documentation'
       ])
     }
+
+    stage('Push to repository') {
+      def secrets = [
```

Comment:
just a personal preference - you can use the much more concise https://github.com/citizensadvice/jenkins-global-pipeline-lib/blob/master/vars/withVaultSecrets.txt

Reply:
I'd forgotten about that. Easy to fix.

Any other issues before I widen out the reviews?

Reply:
looks good to me. The only thing I would try is to leverage some of the internal methods here https://github.com/rubygems/bundler/blob/35be6d9a603084f719fec4f4028c18860def07f6/lib/bundler/gem_helper.rb#L98

e.g. `sh_with_input`

Reply:
..which is protected. :-))